app/coffee/Features/Compile/ClsiManager.coffee

```diff
@@ -7,16 +7,28 @@ ProjectEntityHandler = require("../Project/ProjectEntityHandler")
 logger = require "logger-sharelatex"
 Url = require("url")
 ClsiCookieManager = require("./ClsiCookieManager")
+ClsiStateManager = require("./ClsiStateManager")
 _ = require("underscore")
 async = require("async")
 ClsiFormatChecker = require("./ClsiFormatChecker")
+DocumentUpdaterHandler = require "../DocumentUpdater/DocumentUpdaterHandler"
 
 module.exports = ClsiManager =
 
-	sendRequest: (project_id, user_id, options = {}, callback = (error, status, outputFiles, clsiServerId, validationProblems) ->) ->
+	sendRequest: (project_id, user_id, options = {}, callback) ->
+		ClsiManager.sendRequestOnce project_id, user_id, _.clone(options), (error, status, result...) ->
+			return callback(error) if error?
+			if status is 'conflict'
+				options.state = "conflict" # will force full compile
```

Comment:
Long story: I started that way, but it was slightly ugly as I had to delete it from the options on the full request, because if it got passed through the the CLSI the RequestParser there didn't like it because false wasn't a string.

It's a sign the current method isn't the right one  I think, maybe the following might work better:

compile_type: "full" | "incremental"
state: "hash value"